src/mdm-data-model.model.ts

```diff
@@ -79,4 +80,25 @@ export interface DataModelIntersection {
   intersects: Uuid[];
 }
 
-export type DataModelIntersectionResponse = MdmResponse<DataModelIntersection>;
\ No newline at end of file
+export type DataModelIntersectionResponse = MdmResponse<DataModelIntersection>;
+
+/**
+ * Represents hierarchy information for a Data Model.
+ */
+export interface DataModelHierarchy {
+  /**
+   * The data types available for this Data Model.
+   */
+  dataTypes?: DataTypeDetail[];
+
+  /**
+   * Gets all child Data Classes for this Data Model.
+   */
+  childDataClasses?: DataClassFull[];
+}
```

Comment:
Having 'Hierarchy' in the name seems more appropriate in this context because dataModels cannot be children of other dataModels. However, for consistencies sake (if you do end up renaming the DataClassHierarchy to DataClassNode, then maybe this could become DataModelNode? 

Happy to leave both of these, but at least to me, having 'Node' in the name conveys a sense of position within a larger structure whereas 'hierarchy' makes me think this object represents the entirety of that larger structure.

Reply:
I'm happy with renaming to `DataModelNode`, this does sound better. Actioned.